src/ncp/rcp_host.cpp

```diff
@@ -402,6 +402,12 @@ const char *RcpHost::GetThreadVersion(void)
     return version;
 }
 
+void RcpHost::GetDeviceRole(DeviceRoleHandler aHandler)
```

Comment:
```suggestion
void RcpHost::GetDeviceRole(const DeviceRoleHandler &aHandler)
```
Not sure if this also works in NcpHost use case though.